Project 4/README.md

```diff
@@ -45,3 +45,212 @@ You're probably thinking, "why would I submit a level 8 kata if they're not wort
+
+Title: the politicians from politician families in China.
+
+Description: 
+
+A research team, studying Chinese electoral system, wants to know how many Chinese politician are from politician families. Based on their initial 
+investigation, Chinese electoral system is not as stated in the Chinese constitution, where the government leaders at all levels are elected by the
+people. 
+
+A politician P from politician family is defined as below:
+1) One of P's biological parent is or was a politician; or
+2) One of P's legal parent is or was a politician; or
+3) One of P's spouse or ex-spouse is a politician; or
+4) One of P's spouse's biological parent in law is or was a politician; or
+5) One of P's spouse's legal parent is or was a politican; or
+6) P is a Chinese citizen; and
+7) P was born after 1949.
+
+```
+
+PREFIX wd: <http://www.wikidata.org/entity/>
+PREFIX wdt: <http://www.wikidata.org/prop/direct/>
+PREFIX wikibase: <http://wikiba.se/ontology#>
+
+# politician from family
+
+SELECT (COUNT (?item) AS ?numPoliticianFromFamily) (COUNT (?politician) AS ?totalPolitician)
+
+WHERE {
+  
+  ?item wdt:P31 wd:Q5.          #instance of human.
+  ?item wdt:P27 wd:Q148.        # a citizen of PR of China.  
+  ?item wdt:P106 wd:Q82955.      # politician.
+  {?item wdt:P22 ?father.       
+         ?father wdt:P106 wd:Q82955.}         #item's father is a politician
+  UNION {?item wdt:P25 ?mother.
+               ?mother wdt:P106 wd:Q82955.}   # or item's mother is a politician
+  UNION {?item wdt:P8810 ?parent.
+               ?parent wdt:P106 wd:Q82955.}     #or item's one of parents(other than father and mother defined by P22 and P25).
+  UNION {?item wdt:P3448 ?stepparent.
+               ?stepparent wdt:P106 wd:Q82955.}.  #or item's one of stepparents is a politician. Noting that the content of P25, P8810, P3448 and P106 may overlap. 
+  UNION {?item wdt:P26 ?spouse.
+               ?spouse wdt:P106 wd:Q82955.}     #or item's spouse is a politician.     
+  UNION {?item wdt:P26 ?spouse.
+               ?spouse wdt:P22 ?sFather.
+               ?sFather wdt:P106 wd:Q82955.}      #or item's spouse' father is a politician.
+  UNION {?item wdt:P26 ?spouse.
+               ?spouse wdt:P25 ?sMother.
+               ?sMother wdt:P106 wd:Q82955.}      #or item's spouse' mothe is a politician.
+  UNION {?item wdt:P26 ?spouse.
+               ?spouse wdt:P8810 ?sParent.
+               ?sParent wdt:P106 wd:Q82955.}      #or one of the item's spouse' parent is a politician.
+  UNION {?item wdt:P26 ?spouse.
+               ?spouse wdt:P3448 ?sStepparent.
+               ?sStepparent wdt:P106 wd:Q82955.}. #or one of the item's spouse's stepparent is a politician. 
+  
+  ?item wdt:P569 ?dob.
+  FILTER (?dob >= "1949-10-01"^^xsd:dateTime).   # item is borned later than Jan.1, 1949. 
+  
+  
+  SERVICE wikibase:label { bd:serviceParam wikibase:language "[AUTO_LANGUAGE], zh". } # Helps get the label in your language, if not, then zh language
```

Comment:
Folks now recommend not to use "zh" alone, but instead specify the region or dialect, etc. See here: https://www.w3.org/International/articles/language-tags/